### DISCUSSION TITLE
Identity project (login page) for web apps

### DISCUSSION THREAD

**Discussion by kjartanvalur** [iayti/CleanArchitecture#34] [Ideas]
Can you please add a login page project and use IdentityServer with all the built in features like reset password and  2factor authentication

**Comment by iayti** [comment]
I am considering adding Blazor frontend instead of Client.WorkerService. I am thinking of adding IdentityServer features while adding Blazor. Thank you for your feedback.

Have a nice coding 😊

**Comment by LucaGabi** [comment]
IS4 license is not compatible with MIT I don't think it's a good candidate ..

**Comment by iayti** [comment]
> IS4 license is not compatible with MIT I don't think it's a good candidate ..

I agree with [Answer 1](https://www.quora.com/Can-I-use-Apache-licensed-source-code-and-redistribute-it-as-MIT-license) and [Answer 2](https://www.quora.com/Is-the-MIT-license-compatible-with-the-Apache-License-Version-2-APLv2). They are quite permissive and have only few restrictions.

**Comment by LucaGabi** [comment]
From what I know IS4 will no longer be maintained:

https://github.com/IdentityServer/IdentityServer4

```
As of Oct 1st 2020, we started a new company. All new feature work will happen in our new organization. 
The new Duende IdentityServer is available under both a FOSS (RPL) and a commercial license. 
Development and testing is always free. Contact us for more information. 
This repo will be maintained with bug fixes and security updates until November 2022.
```

**Comment by iayti** [comment]
I haven't been looking at the IS4 repository for a long time. Thank you for the information. I just check the license compatibility for the current versions.

I think it is not a good idea for using IS4 on an open-source project. If there is any tool like IS4 with an open-source licence, we can implement it. I am open to advice 😊 

We can also implement our own identity service.

**Comment by wengtad** [comment]
How about openiddict-core by OpenIddict? But their license is [Apache-2.0 License](https://github.com/openiddict/openiddict-core/blob/dev/LICENSE.md) not sure compatible with MIT or not. I'm not that sure about license.

**Reply by iayti** [reply]
Thank you for your advice. I will check [OpenIddict](https://github.com/openiddict/openiddict-core). I don't think the license(Apache-2.0 License) will be a problem.
